src/main/kotlin/org/rucca/cheese/notification/NotificationController.kt

```diff
@@ -6,63 +6,97 @@ import org.rucca.cheese.auth.AuthenticationService
 import org.rucca.cheese.auth.AuthorizationService
 import org.rucca.cheese.auth.AuthorizedAction
 import org.rucca.cheese.auth.annotation.Guard
+import org.rucca.cheese.auth.annotation.ResourceId
 import org.rucca.cheese.common.persistent.IdGetter
 import org.rucca.cheese.common.persistent.IdType
 import org.rucca.cheese.model.*
-import org.springframework.http.HttpStatus
+import org.rucca.cheese.user.UserService
 import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.*
 
 @RestController
 class NotificationController(
     private val notificationService: NotificationService,
+    private val userService: UserService,
     private val authorizationService: AuthorizationService,
     private val authenticationService: AuthenticationService,
 ) : NotificationsApi {
 
     @PostConstruct
     fun initialize() {
         authorizationService.ownerIds.register(
-            "notifications",
+            "notification",
             notificationService::getNotificationOwner,
         )
-        authorizationService.customAuthLogics.register("is-notification-admin") {
+
+        authorizationService.customAuthLogics.register("is-notification-owner") {
             userId: IdType,
-            _: AuthorizedAction,
+            action: AuthorizedAction,
             _: String,
             resourceId: IdType?,
             _: Map<String, Any?>?,
             _: IdGetter?,
             _: Any? ->
-            notificationService.isNotificationAdmin(
-                resourceId ?: throw IllegalArgumentException("resourceId is null"),
-                userId,
-            )
+            when (action) {
+                "delete" -> {
+                    if (resourceId == null) {
+                        return@register false
+                    }
+                    return@register notificationService.isNotificationAdmin(resourceId, userId)
+                }
+                else -> {
+                    return@register true
+                }
+            }
         }
     }
 
-    //  unused function
-    fun createNotification(notification: Notification) {
-        notificationService.createNotification(notification)
+    @Guard("create", "notification")
+    override fun postNotification(
+        postNotificationRequestDTO: PostNotificationRequestDTO
+    ): ResponseEntity<PostNotification200ResponseDTO> {
+        if (!userService.existsUser(postNotificationRequestDTO.receiverId)) {
```

Comment:
Should be inside the service instead of the controller. Since it's about actual business-logic.